### PR TITLE
[Doc] Add developer notes

### PR DESCRIPTION
**Pull request by riedgar-ms**
Expand the CONTRIBUTING file to describe the basic developer setup. Since our workflows access some quite expensive resources, PRs have to be approved to run. This will let contributors get feedback faster.

### CODE REVIEW

**Comment by Harsha-Nori** [comment]
Thoughts on extending `CONTRIBUTING.MD` with this info instead of having a new top level file here?

**Comment by riedgar-ms** [comment]
> Thoughts on extending `CONTRIBUTING.MD` with this info instead of having a new top level file here?

That's an option too. I thought about it, but the existing file was more 'legalese' than information on how to contribute. Happy to switch if you prefer.

File: DevelopmentNotes.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+```
+which runs our basic test suite.
+Where an LLM is required, this will default to using GPT2 on the CPU.
+To change that default, run
+```bash
+python -m pytest -m "not (needs_credentials or use_gpu or server)" --selected_model <MODELNAME> ./tests/
+```
+where `<MODELNAME>` is taken from the `AVAILABLE_MODELS` dictionary defined in `conftest.py`.
+
+## Linting
```
Comment (riedgar-ms) [inline DevelopmentNotes.md:29]: Need to add a section on adding models to the test setup

**Comment by riedgar-ms** [comment]
@Harsha-Nori , I've moved everything into CONTRIBUTING. I've also added some notes about how to hook new models into the testing setup on the off chance that random people start pushing to `main`.

**Pull request merged**